ultra/competition/starting_kit/README.md

```diff
@@ -0,0 +1,372 @@
+        observation will be the adapted observation, that is, it will be what is
+        returned from the agent's observation adapter.
+
+    Returns:
+      Union[Sequence[float], np.ndarray]: Either a Sequence of three float values, or a
+        NumPy array with shape (3,). The action is the [throttle, brake, steering] of
+        the agent's vehicle.
+    """
+    pass
+```
+
+The agent must contain an act method that returns an action given an observation.
+
+### The AgentSpec
+
+In addition to the agent class, a `smarts.core.agent.AgentSpec` will be needed. The `AgentSpec` defines how this agent class will be instantiated and how it will interact with the environment.
+
+The `AgentSpec` takes a few specific parameters:
+
+- `interface`:
+
+  The `AgentInterface` provides the SMARTS simulator with information about what sensors the vehicle needs and also what action the agent outputs. Recall that we restrict the agent's observation to either be the "vector" or "image" type, and that the agent should be using a continuous action space. So depending on the type of observation you would like to use, your `AgentInterface` will look like:
+
+  ```python
+  from smarts.core.controllers import ActionSpaceType
+  from smarts.core.agent_interface import AgentInterface, NeighborhoodVehicles, RGB, Waypoints
+
+  # If using the image observation...
+  AgentInterface(
+    rgb=RGB(width=64, height=64, resolution=(50 / 64)),
+    action=ActionSpaceType.Continuous,
+  )
+
+  # If using the vector observation...
+  AgentInterface(
+    neighborhood_vehicles=NeighborhoodVehicles(radius=200.0),
+    waypoints=Waypoints(lookahead=20),
+    action=ActionSpaceType.Continuous,
+  )
+  ```
+
+  The image observation adapter requires the car to have an RGB sensor that produces an image with width of 64 pixels and height of 64 pixels. The resolution of the image is also recommended to be 50 / 64. That is, the sensor will produce a 64x64 image that spans 50 meters by 50 meters. The values of 64, 64, and 50 / 64 for the `width`, `height`, and `resolution`, respectively, are what will be used for the evaluation on CodaLab, so it is recommended that these values be used. If your agent does not have these values for the `width`, `height`, and `resolution` when submitted for evaluation on CodaLab, the evaluation will fail.
+
+  The vector observation adapter requires the car to have a NeighborhoodVehicles sensor that senses all nearby vehicles within a certain `radius`, and a Waypoints sensor that produces waypoints up to a certain `lookahead` distance along the vehicle's mission path. The values of 200.0 and 20 for the `radius` and `lookahead`, respectively, can
+  be changed. However, these are the values that will be used for the evaluation on CodaLab, so it is recommended that these values be used. If your agent does not have these values for the `radius` and `lookahead` when sumitted for evaluation on CodaLab, the evaluation will fail.
```

Comment:
If we're going with all text on one line, I think these "paragraphs" should be together? (also typo in 'sumitted')
```suggestion
  The vector observation adapter requires the car to have a NeighborhoodVehicles sensor that senses all nearby vehicles within a certain `radius`, and a Waypoints sensor that produces waypoints up to a certain `lookahead` distance along the vehicle's mission path. The values of 200.0 and 20 for the `radius` and `lookahead`, respectively, can be changed. However, these are the values that will be used for the evaluation on CodaLab, so it is recommended that these values be used. If your agent does not have these values for the `radius` and `lookahead` when submitted for evaluation on CodaLab, the evaluation will fail.
```